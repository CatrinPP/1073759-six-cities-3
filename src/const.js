import PropTypes from 'prop-types';

const CITIES_LIST_COUNT = 6;
const MAX_COMMENTS_VALUE = 10;
const MAP_ICON_SIZE = 30;
const MAX_OFFERS_NEARBY = 3;
const MAP_SIZE_DETAILED_OFFER = `1144px`;
const TIMEOUT = 5000;
const ZOOM_VALUE = 12;

const PlaceType = {
  APARTMENT: `Apartment`,
  BUNGALOW: `Bungalow`,
  HOUSE: `House`,
  ROOM: `Room`,
  STUDIO: `Studio`,
  VILLA: `Villa`
};

const SortingType = {
  DEFAULT: `Popular`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`
};

const sortTypes = [SortingType.DEFAULT, SortingType.PRICE_LOW_TO_HIGH, SortingType.PRICE_HIGH_TO_LOW, SortingType.TOP_RATED];

const Feature = {
  WI_FI: `Wifi`,
  HEATING: `Heating`,
  KITCHEN: `Kitchen`,
  CABLE_TV: `Cable TV`,
  FRIDGE: `Fridge`,
  WASHING_MACHINE: `Washing machine`,
  COFFEE_MACHINE: `Coffee machine`,
  DISHWASHER: `Dishwasher`,
  TOWELS: `Towels`,
  BABY_SEAT: `Baby seat`
};

const cityShape = {
  name: PropTypes.string.isRequired,
  coords: PropTypes.arrayOf(PropTypes.number.isRequired),
};

const offerShape = {
  bedrooms: PropTypes.number,
  coords: PropTypes.arrayOf(PropTypes.number.isRequired),
  description: PropTypes.string,
  features: PropTypes.arrayOf(PropTypes.oneOf([Feature.WI_FI, Feature.HEATING, Feature.KITCHEN, Feature.CABLE_TV, Feature.FRIDGE, Feature.WASHING_MACHINE, Feature.COFFEE_MACHINE, Feature.DISHWASHER, Feature.TOWELS, Feature.BABY_SEAT])),
  guests: PropTypes.number,
  host: PropTypes.shape({
    avatar: PropTypes.string,
    name: PropTypes.string,
    isStar: PropTypes.bool,
  }),
  id: PropTypes.string.isRequired,
  images: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  isPremium: PropTypes.bool.isRequired,
  price: PropTypes.number.isRequired,
  rating: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.oneOf([PlaceType.APARTMENT, PlaceType.BUNGALOW, PlaceType.HOUSE, PlaceType.ROOM, PlaceType.STUDIO, PlaceType.VILLA]).isRequired,
};

const offersListShape = {
  city: PropTypes.shape(cityShape).isRequired,
  coords: PropTypes.arrayOf(PropTypes.number.isRequired),
  offers: PropTypes.arrayOf(PropTypes.shape(offerShape)),
};

const userShape = {
  avatar: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

const commentShape = {
  date: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  rating: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
  user: PropTypes.shape(userShape).isRequired
};

export {
  CITIES_LIST_COUNT,
  cityShape,
  commentShape,
  MAX_COMMENTS_VALUE,
  MAP_ICON_SIZE,
  MAX_OFFERS_NEARBY,
  MAP_SIZE_DETAILED_OFFER,
  offersListShape,
  offerShape,
  SortingType,
  sortTypes,
  TIMEOUT,
  ZOOM_VALUE,
};
