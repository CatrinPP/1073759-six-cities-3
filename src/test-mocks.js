const testFavorites = [
  {
    city: `Moscow`,
    offers: [
      {
        bedrooms: 2,
        coords: [52.3909553943508, 4.85309666406198],
        description: ``,
        features: [`Washer`, `Washing machine`],
        guests: 5,
        host: {
          avatar: ``,
          name: `Ann`,
          isStar: true
        },
        id: 1,
        images: [`img/room.jpg`, `img/room1.jpg`, `img/room2.jpg`],
        isFavorite: false,
        isPremium: true,
        price: 98,
        rating: 0,
        title: `Cozy Studio`,
        type: `room`,
      }, {
        bedrooms: 4,
        coords: [52.369553943508, 4.85309666406198],
        description: ``,
        features: [`Fridge`],
        guests: 10,
        host: {
          avatar: ``,
          name: `Anny`,
          isStar: false
        },
        id: 2,
        images: [`img/room3.jpg`, `img/room4.jpg`, `img/room5.jpg`],
        isFavorite: false,
        isPremium: false,
        price: 34,
        rating: 50,
        title: `Big Vintage Apartment`,
        type: `apartment`,
      },
    ]
  }, {
    city: `Tokyo`,
    offers: [
      {
        bedrooms: 1,
        coords: [52.3909553943508, 4.929309666406198],
        description: ``,
        features: [`Cable TV`],
        guests: 1,
        host: {
          avatar: ``,
          name: `Vince`,
          isStar: true
        },
        id: 3,
        images: [`img/room6.jpg`, `img/room7.jpg`, `img/room8.jpg`],
        isFavorite: false,
        isPremium: false,
        price: 80,
        rating: 100,
        title: `Small Country House`,
        type: `house`,
      }, {
        bedrooms: 0,
        coords: [52.3809553943508, 4.939309666406198],
        description: ``,
        features: [`Breakfast`],
        guests: 0,
        host: {
          avatar: ``,
          name: `Minni`,
          isStar: false
        },
        id: 4,
        images: [`img/room9.jpg`, `img/room10.jpg`, `img/room11.jpg`],
        isFavorite: false,
        isPremium: true,
        price: 3200,
        rating: 95,
        title: `Villa with no view`,
        type: `hotel`,
      }, {
        bedrooms: 20,
        coords: [52.3809553943508, 4.93930966640619],
        description: ``,
        features: [`Breakfast`, `Washer`],
        guests: 50,
        host: {
          avatar: ``,
          name: `Polly`,
          isStar: true
        },
        id: 5,
        images: [`img/room12.jpg`, `img/room13.jpg`, `img/room14.jpg`],
        isFavorite: false,
        isPremium: false,
        price: 5,
        rating: 15,
        title: `Tiny bungalow`,
        type: `house`,
      },
    ]
  }
];


const testOffers = [
  {
    city: {
      name: `Moscow`,
      coords: [44.44, 2.22],
    },
    offers: [
      {
        bedrooms: 2,
        coords: [52.3909553943508, 4.85309666406198],
        description: ``,
        features: [`Washer`, `Washing machine`],
        guests: 5,
        host: {
          avatar: ``,
          name: `Ann`,
          isStar: true
        },
        id: 1,
        images: [`img/room.jpg`, `img/room1.jpg`, `img/room2.jpg`],
        isFavorite: false,
        isPremium: true,
        price: 98,
        rating: 0,
        title: `Cozy Studio`,
        type: `room`,
      }, {
        bedrooms: 4,
        coords: [52.369553943508, 4.85309666406198],
        description: ``,
        features: [`Fridge`],
        guests: 10,
        host: {
          avatar: ``,
          name: `Anny`,
          isStar: false
        },
        id: 2,
        images: [`img/room3.jpg`, `img/room4.jpg`, `img/room5.jpg`],
        isFavorite: false,
        isPremium: false,
        price: 34,
        rating: 50,
        title: `Big Vintage Apartment`,
        type: `apartment`,
      },
    ]
  }, {
    city: {
      name: `Tokyo`,
      coords: [144.44, 5.22],
    },
    offers: [
      {
        bedrooms: 1,
        coords: [52.3909553943508, 4.929309666406198],
        description: ``,
        features: [`Cable TV`],
        guests: 1,
        host: {
          avatar: ``,
          name: `Vince`,
          isStar: true
        },
        id: 3,
        images: [`img/room6.jpg`, `img/room7.jpg`, `img/room8.jpg`],
        isFavorite: false,
        isPremium: false,
        price: 80,
        rating: 100,
        title: `Small Country House`,
        type: `house`,
      }, {
        bedrooms: 0,
        coords: [52.3809553943508, 4.939309666406198],
        description: ``,
        features: [`Breakfast`],
        guests: 0,
        host: {
          avatar: ``,
          name: `Minni`,
          isStar: false
        },
        id: 4,
        images: [`img/room9.jpg`, `img/room10.jpg`, `img/room11.jpg`],
        isFavorite: false,
        isPremium: true,
        price: 3200,
        rating: 95,
        title: `Villa with no view`,
        type: `hotel`,
      }, {
        bedrooms: 20,
        coords: [52.3809553943508, 4.93930966640619],
        description: ``,
        features: [`Breakfast`, `Washer`],
        guests: 50,
        host: {
          avatar: ``,
          name: `Polly`,
          isStar: true
        },
        id: 5,
        images: [`img/room12.jpg`, `img/room13.jpg`, `img/room14.jpg`],
        isFavorite: false,
        isPremium: false,
        price: 5,
        rating: 15,
        title: `Tiny bungalow`,
        type: `house`,
      },
    ]
  }, {
    city: {
      name: `Oslo`,
      coords: [14.44, 52.22],
    },
    offers: [
      {
        bedrooms: 2,
        coords: [14.3909553943508, 52.929309666406198],
        description: ``,
        features: [`Fridge`, `Baby seat`],
        guests: 3,
        host: {
          avatar: ``,
          name: `Vincena`,
          isStar: true
        },
        id: 6,
        images: [`img/room15.jpg`, `img/room16.jpg`, `img/room17.jpg`],
        isFavorite: false,
        isPremium: true,
        price: 180,
        rating: 40,
        title: `Very Small Room`,
        type: `room`,
      }, {
        bedrooms: 3,
        coords: [15.3809553943508, 52.939309666406198],
        description: ``,
        features: [`Air conditioning`],
        guests: 5,
        host: {
          avatar: ``,
          name: `Mi`,
          isStar: false
        },
        id: 7,
        images: [`img/room18.jpg`, `img/room19.jpg`, `img/room20.jpg`],
        isFavorite: false,
        isPremium: true,
        price: 32,
        rating: 75,
        title: `Room without windows`,
        type: `room`,
      },
    ]
  }, {
    city: {
      name: `Vatican`,
      coords: [33.44, 12.22],
    }
  }
];

const testComments = [
  {
    date: `2018-05-08T14:13:56.569Z`,
    id: 1,
    rating: 4,
    text: `Test Comment #1`,
    user: {
      avatar: `img/avatar-max.jpg`,
      name: `Max`,
    }
  }, {
    date: `2018-05-09T14:13:56.569Z`,
    id: 2,
    rating: 3,
    text: `Test Comment #2`,
    user: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Angelina`,
    }
  }, {
    date: `2018-09-10T14:13:56.569Z`,
    id: 3,
    rating: 4,
    text: `Test Comment #3`,
    user: {
      avatar: `img/avatar-max.jpg`,
      name: `Max`,
    }
  }, {
    date: `2018-01-11T14:13:56.569Z`,
    id: 4,
    rating: 3,
    text: `Test Comment #4`,
    user: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Angelina`,
    }
  }, {
    date: `2018-10-12T14:13:56.569Z`,
    id: 5,
    rating: 4,
    text: `Test Comment #5`,
    user: {
      avatar: `img/avatar-max.jpg`,
      name: `Max`,
    }
  }, {
    date: `2018-05-13T14:13:56.569Z`,
    id: 6,
    rating: 3,
    text: `Test Comment #6`,
    user: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Angelina`,
    }
  }, {
    date: `2018-11-18T14:13:56.569Z`,
    id: 7,
    rating: 4,
    text: `Test Comment #7`,
    user: {
      avatar: `img/avatar-max.jpg`,
      name: `Max`,
    }
  }, {
    date: `2019-01-01T14:13:56.569Z`,
    id: 8,
    rating: 3,
    text: `Test Comment #8`,
    user: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Angelina`,
    }
  },
];

export {testFavorites, testOffers, testComments};
