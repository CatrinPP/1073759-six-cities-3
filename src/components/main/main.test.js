import React from 'react';
import {Router} from "react-router-dom";
import history from "../../history.js";
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from "redux-mock-store";
import Main from './main.jsx';
import {testOffers} from '../../test-mocks.js';
import {SortingType, AuthorizationStatus} from '../../const.js';
import NameSpace from '../../reducer/name-space.js';

const mockStore = configureStore([]);

const mockCity = {
  name: `Paris`,
};

it(`Should render Main with not yet loaded offers correctly`, () => {
  const store = mockStore({
    [NameSpace.APP]: {
      city: mockCity,
      sortType: SortingType.DEFAULT,
    },
    [NameSpace.DATA]: {
      allOffers: testOffers,
      isLoaded: false,
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
    },
  });

  const tree = renderer
    .create(
        <Router history={history}>
          <Provider store={store}>
            <Main />
          </Provider>
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Should render Main with offers correctly`, () => {
  const store = mockStore({
    [NameSpace.APP]: {
      city: mockCity,
      sortType: SortingType.DEFAULT,
    },
    [NameSpace.DATA]: {
      allOffers: testOffers,
      isLoaded: true,
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
    },
  });

  const tree = renderer
    .create(
        <Router history={history}>
          <Provider store={store}>
            <Main />
          </Provider>
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Should render Main for authorized users correctly`, () => {
  const store = mockStore({
    [NameSpace.APP]: {
      city: mockCity,
      sortType: SortingType.DEFAULT,
    },
    [NameSpace.DATA]: {
      allOffers: testOffers,
      isLoaded: true,
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.AUTH,
    },
  });

  const tree = renderer
    .create(
        <Router history={history}>
          <Provider store={store}>
            <Main />
          </Provider>
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
