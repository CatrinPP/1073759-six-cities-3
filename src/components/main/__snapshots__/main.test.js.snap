// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Main correctly 1`] = `""`;
